algorithm-prototyping/src/main/resources/scripts/generate_distance_matrix.py

```diff
@@ -0,0 +1,28 @@
+import json
+
+# Read buildings.json and store data as a dictionary
+with open('buildings.json', 'r') as file:
+    data = json.load(file)
+
+# Initialize output dictionary with buildings array and distance matrix
+distanceData = {
+    'distance_matrix': {}
+}
+
+#Calculate distance matrix and update output dictionary
+for building1 in data['buildings']:

```

Comment:
Should be `for building1 in data:`. See format for `buildings.json`:

```json
[
  {
   "name": "Park Hall",
    "x": 3,
    "y": 2
  },
  ...
]
```

Script works well with this change made.